src/main/java/com/connexta/search/common/advice/SearchExceptionHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.common.advice;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+import com.connexta.search.common.exceptions.DetailedErrorAttributes;
+import com.connexta.search.index.exceptions.IndexException;
+import com.connexta.search.query.exceptions.QueryException;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.validation.ConstraintViolationException;
+import javax.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+@Slf4j
+public class SearchExceptionHandler extends ResponseEntityExceptionHandler {
+
+  @Inject private DetailedErrorAttributes detailedErrorAttributes;
+
+  @ExceptionHandler(ConstraintViolationException.class)
+  protected ResponseEntity<Object> handleConstraintViolation(
+      @NotNull final ConstraintViolationException e, @NotNull final WebRequest request) {
+    log.warn("Request is invalid: {}. Returning {}.", e.getMessage(), BAD_REQUEST, e);
+    Map<String, Object> serializableError =
+        detailedErrorAttributes.getErrorAttributes(request, false);
+    return new ResponseEntity<>(serializableError, BAD_REQUEST);
+  }
+
+  @ExceptionHandler(QueryException.class)
+  protected ResponseEntity<Object> handleQueryException(
+      @NotNull final QueryException e, final WebRequest request) {
+    log.warn("Query exception", e);
+    return new ResponseEntity<>(e, e.getStatus());
+  }
```

Comment:
✏️ Move this method to a class in the `com.connexta.search.query` package because it is not common. Same for the method below.

Reply:
That would mean creating a new class. For one method. If we have plans to break up this repo, I can see the point of it. I like the idea of having all the error handlers for the application in one place.